src/python/visclaw/plotpages.py

```diff
@@ -549,7 +545,242 @@ def plots2latex(plot_pages_data):
+            fname = 'frame' + str(frameno).rjust(4, '0')
+            fname_str = fname + 'fig%s' % figno
+
+            print '\n'
+            print "Tiling %s.png and adding KML entry to doc.kml" % (fname_str)
+
+            doc.Document.append(
+                KML.NetworkLink(
+                    KML.name(fname_str),
+                    KML.TimeSpan(
+                        KML.begin(timestrbegin),
+                        KML.end(timestrend)),
+                        KML.LatLonBox(
+                            KML.north(ur[1]),KML.south(lr[1]),KML.east(ul[0]),KML.west(ur[0]),
+                            KML.rotation(0.0)),
+                    KML.Link(
+                        KML.href("%s/doc.kml" % fname_str))))
+
+            im = plt.imread("%s.png" % fname_str)
+            sx = im.shape[1]   # reversed?
+            sy = im.shape[0]
+
+            # gdal_str = "gdal_translate -of VRT " \
+            #        "-a_srs EPSG:4326 " \
+            #        "-gcp %d     %d  %f   %f      " \
+            #        "-gcp %d     %d  %f   %f      " \
+            #        "-gcp %d     %d  %f   %f  -90 " \
+            #        "%s.png %s_tmp.vrt" %(0,0,ul[0],ul[1],
+            #                              sx,0,ur[0],ur[1],
+            #                              sx,sy,lr[0],lr[1],
+            #                              fname_str,fname_str)
+            # os.system(gdal_str)
+            # os.system("gdalwarp -of VRT -t_srs EPSG:4326 " \
+            #           "-overwrite %s_tmp.vrt %s.vrt" % (fname_str,fname_str))
+            #
+            # os.system("gdal2tiles.py " \
+            #           "--profile=geodetic  " \
+            #           "--force-kml " \
+            #           "--resampling=near " \
+            #           "%s.vrt" % (fname_str))
+
+            arg_list = ["gdal_translate", "-of", "VRT", \
+                        "-a_srs", "EPSG:4326",  \
+                        "-gcp", "0",         "0",        "%f"%(ul[0]),   "%f"%(ul[1]), \
+                        "-gcp", "%d"%(sx),   "0",        "%f"%(ur[0]),   "%f"%(ur[1]), \
+                        "-gcp", "%d"%(sx),   "%d"%(sy),  "%f"%(lr[0]),   "%f"%(lr[1]), "-90", \
+                        "%s.png"%(fname_str), "%s_tmp.vrt"%(fname_str)];
+            import subprocess
+            retval = subprocess.call(arg_list)
+
+            arg_list = ["gdalwarp", "-of", "VRT", "-t_srs", "EPSG:4326 ", \
+                        "-overwrite", "%s_tmp.vrt"%(fname_str), "%s.vrt"%(fname_str)]
+            retval = retval or subprocess.call(arg_list)
+
+            arg_list = ["gdal2tiles.py", \
+                       "--profile=geodetic", \
+                       "--force-kml", \
+                       "--resampling=near", \
+                       "%s.vrt" % (fname_str)]
+            retval = retval or subprocess.call(arg_list)
+
+            if retval > 0:
+                print "gdal : something went wrong!\n"
+                sys.exit(1)
```

Comment:
Probably better to raise an appropriate exception here rather than just exiting.